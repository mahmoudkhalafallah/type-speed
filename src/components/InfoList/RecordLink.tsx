import React from 'react'
import { Link } from '@reach/router'
const RecordLink: React.FC<{ historyId: string }> = ({ historyId }) =>
    <>
        Find your record <Link to={`/history/${historyId}`} data-testid="recordLink">here</Link>
    </>
export default RecordLink