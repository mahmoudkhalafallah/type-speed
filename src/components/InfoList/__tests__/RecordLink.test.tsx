import React from 'react'
import RecordLink from '../RecordLink'
import { render } from '@testing-library/react'

const testId = 'testId'

test('RecordLink rendered correctly', () => {
    const { baseElement, queryByTestId } = render(
        <RecordLink
            historyId={testId}
        />,
    )
    expect(queryByTestId('recordLink')).toHaveAttribute('href', `/history/${testId}`)
    expect(baseElement).toMatchSnapshot()
})