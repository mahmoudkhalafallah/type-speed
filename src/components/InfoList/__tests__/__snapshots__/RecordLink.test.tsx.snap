// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecordLink rendered correctly 1`] = `
<body>
  <div>
    Find your record 
    <a
      data-testid="recordLink"
      href="/history/testId"
    >
      here
    </a>
  </div>
</body>
`;
